index.js

```diff
@@ -109,11 +116,47 @@ module.exports = fp(
                   bufferFlushTimeout,
                   udpDnsCache,
                   udpDnsCacheTTL,
-                  onError: onError,
+                  onError,
               })
             : clientMock();
 
         fastify.decorate('stats', stats);
+
+        const _onSend = defaultOnSend.bind(fastify);
+        fastify.decorate(
+            'timerify',
+            function (name, fn, onSend = _onSend, opts) {
```

Comment:
It's ok we don't need to change this since it's already done but can we talk about this pattern of making options/callback be interchangeable for the future? Is it really needed? It's been used from the dawn of time but I would make them required and always in the same order, less boilerplate, and easier to read! wdyt?

Reply:
I agree, but in this case the `opts` argument can be passed only on Node versions from 16 and newer. It also is an optional argument. It's an option object passed to the core timerify function. I couldn't find a better way atm. I didn't mark them as required because I didn't want to export the default onSend function and to pass an opts option you also have to pass  the onSend parameter, even if you didn't want to change it.